src/processors/chain/ProcessorChainActionHelper.cpp

```diff
@@ -46,9 +48,44 @@ void ProcessorChainActionHelper::addProcessor (BaseProcessor::Ptr newProc)
         jassertfalse; // unable to create this processor!
         return;
     }
+    else
+    {
+        um->beginNewTransaction();
+        um->perform (new AddOrRemoveProcessor (chain, std::move (newProc)));
+    }
+}
 
+void ProcessorChainActionHelper::addProcessorFromCableClick (BaseProcessor::Ptr newProc, BaseProcessor* startProc, BaseProcessor* endProc)
+{
     um->beginNewTransaction();
+
+    // Destroy old cable connection
+    auto minIncomingPorts = jmin (endProc->getNumInputs(), startProc->getNumOutputs());
+    for (int portIdx = 0; portIdx < minIncomingPorts; ++portIdx)
+    {
+        um->perform (new AddOrRemoveConnection (chain, { startProc, portIdx, endProc, portIdx }, true));
```

Comment:
Hmm, so I think this logic might be a little bit too agressive... for example, if the user has the Stereo Splitter/Stereo Merger modules connected to each other with two cables, I think this logic would either remove all the connections (if port 0 is connected to port 0, and port 1 -> port 1), or remove none of the connections (port 0 -> port 1, port 1 -> port 0). Or if only one set of inputs/outputs is connected, I think this logic would try to remove a connection that doesn't exist.

I think a potential solution (which might also make the upstream code a bit cleaner) would be to pass a `Connection` object rather than two `BaseProcessor` objects to this method. That way this logic will know exactly which connection should be removed, and will know with certainty that the connection exists.

Reply:
This makes sense, I did some testing with the Splitter and Merger and it did mess with the current logic so thanks for seeing that.

I'm just looking into replacing the two processors with connection, and I think that should make the logic more bulletproof. The only thing I'm not sure about is all the default arguments and `nullptr` checks that would need to be changed. if a `ConnectionInfo` object was passed through instead of the two `BaseProcessor* `'s. For example `createProcListFiltered()` is currently set up to take a startProc and endProc, and we could just pass a `nullptr` when not using those arguments, but I'm not totally sure the best way to approach that problem with a `ConnectionInfo` instead of the two processors.

I'm sure there's a good way to do it, just thought I would ask if you had any recommendations before I looked into it too deeply.

Thanks!

Reply:
Yeah, that's a good question... I think it should probably start from `createProcFromCableClickList`. Like the signature for that method could be something like `ProcessorStore::createProcFromCableClickList (PopupMenu& menu, int& menuID, ConnectionInfo& connectionInfo)`. Then the connection info could be passed down the callstack. And then you'd probably have to convert between using a reference and pointer whenever there's a default argument that needs to be treated that way.

Reply:
That helps a lot! I'll look into changing those functions around thanks.